File: scripts/setup-docuseal-template.py
#!/usr/bin/env python3
"""
One-time setup script to create a reusable DocuSeal template for RTR documents
Run: python3 scripts/setup-docuseal-template.py
"""

import os
import sys
import base64
import json
import requests
from pathlib import Path

def create_rtr_template():
    # Read API key from environment or .env file
    api_key = os.environ.get("DOCUSEAL_API_KEY")

    if not api_key:
        # Try reading from supabase/functions/.env
        env_file = Path("supabase/functions/.env")
        if env_file.exists():
            with open(env_file) as f:
                for line in f:
                    if line.startswith("DOCUSEAL_API_KEY="):
                        api_key = line.split("=", 1)[1].strip()
                        break

    if not api_key:
        print("❌ DOCUSEAL_API_KEY not found")
        print("Make sure it's set in supabase/functions/.env")
        sys.exit(1)

    base_url = "https://api.docuseal.com"

    print("🔧 Creating DocuSeal RTR template...")
    print()

    # Create a simple HTML-based template
    # DocuSeal will render this as a PDF and allow field placement
    html_content = """
    <!DOCTYPE html>
    <html>
    <head>
        <meta charset="UTF-8">
        <style>
            body { font-family: Arial, sans-serif; margin: 40px; }
            h1 { text-align: center; }
            .section { margin: 20px 0; }
        </style>
    </head>
    <body>
        <h1>Right to Represent Agreement</h1>

        <div class="section">
            <p><strong>This document will be replaced with your actual RTR template.</strong></p>
            <p>After creating this template, you'll need to:</p>
            <ol>
                <li>Go to DocuSeal dashboard</li>
                <li>Find this template</li>
                <li>Upload your actual RTR PDF document</li>
                <li>Add signature and form fields</li>
            </ol>
        </div>

        <div class="section">
            <p>Candidate Signature: _____________________</p>
            <p>Date: _____________________</p>
        </div>
    </body>
    </html>
    """

    payload = {
        "name": "RTR Document Template (CompSciPrep)",
        "html": html_content
    }

    response = requests.post(
        f"{base_url}/templates/html",
        headers={
            "X-Auth-Token": api_key,
            "Content-Type": "application/json"
        },
        json=payload
    )

    if not response.ok:
        print(f"❌ Failed to create template ({response.status_code}):")
        print(response.text)
        sys.exit(1)

    template = response.json()
    template_id = str(template.get("id") or template.get("slug"))

    print()
    print("✅ Template created successfully!")
    print()
    print(f"📋 Template ID: {template_id}")
    print()
    print("🔗 Next steps:")
    print(f"   1. Go to: https://www.docuseal.com/templates/{template_id}")
    print(f"   2. Replace the HTML with your actual RTR PDF (upload document)")
    print(f"   3. Add signature and form fields using the visual editor")
    print(f"   4. Set role as 'Candidate' for all fields")
    print()
    print("⚙️  Add this to your supabase/functions/.env file:")
    print()
    print(f"DOCUSEAL_RTR_TEMPLATE_ID={template_id}")
    print()
    print("Then restart your edge functions server")
    print()

if __name__ == "__main__":
    try:
        create_rtr_template()
    except Exception as e:
        print(f"❌ Error: {e}")
        sys.exit(1)

File: supabase/functions/send-rtr-email/setup-docuseal-template.ts
/**
 * One-time setup script to create a reusable DocuSeal template for RTR documents
 *
 * Run with: deno run --allow-net --allow-env --allow-read setup-docuseal-template.ts
 */

import { bytesToBase64 } from "./docuseal.ts";

async function createRtrTemplate() {
  // Read environment variables
  const apiKey = Deno.env.get("DOCUSEAL_API_KEY");
  if (!apiKey) {
    console.error("❌ DOCUSEAL_API_KEY not set");
    console.error("Set it with: export DOCUSEAL_API_KEY=your_key");
    Deno.exit(1);
  }

  const baseUrl = (Deno.env.get("DOCUSEAL_BASE_URL") || "https://api.docuseal.com").replace(/\/$/, "");

  // Read the sample RTR PDF (we need a PDF to create the template)
  console.log("📄 Checking for RTR template PDF...");
  const docxPath = Deno.env.get("RTR_TEMPLATE_DOCX_PATH") || "docs/CompSciPrep_RTR_Template.docx";

  console.log(`⚠️  Note: This script needs a sample RTR PDF to create the template.`);
  console.log(`   You have two options:`);
  console.log(`   1. Generate a test RTR PDF first by running the RTR flow once`);
  console.log(`   2. Convert the DOCX template to PDF manually and save it`);
  console.log(``);
  console.log(`   For now, I'll create a basic template that you'll configure in DocuSeal UI.`);
  console.log(``);

  // Create a minimal template without PDF upload
  // DocuSeal allows creating templates that you complete in their UI
  console.log("🔧 Creating DocuSeal template...");

  const templatePayload = {
    name: "RTR Document Template (CompSciPrep)",
    external_id: "rtr-compsciprep-main",
  };

  const response = await fetch(`${baseUrl}/templates`, {
    method: "POST",
    headers: {
      "X-Auth-Token": apiKey,
      "Content-Type": "application/json",
    },
    body: JSON.stringify(templatePayload)
  });

  if (!response.ok) {
    const errorText = await response.text();
    console.error(`❌ Failed to create template (${response.status}):`, errorText);
    Deno.exit(1);
  }

  const template = await response.json();
  const templateId = template.id || template.slug;

  console.log(``);
  console.log("✅ Template created successfully!");
  console.log(``);
  console.log("📋 Template ID:", templateId);
  console.log(``);
  console.log("🔗 Configure your template:");
  console.log(`   1. Go to: https://www.docuseal.com/templates/${templateId}`);
  console.log(`   2. Upload your RTR PDF document`);
  console.log(`   3. Add signature and form fields using the visual editor`);
  console.log(`   4. Set role as "Candidate" for all fields`);
  console.log(``);
  console.log("⚙️  Add this to your supabase/functions/.env file:");
  console.log(``);
  console.log(`DOCUSEAL_RTR_TEMPLATE_ID=${templateId}`);
  console.log(``);
  console.log("🔄 After configuring, restart your edge functions server");
}

// Run the setup
createRtrTemplate().catch((error) => {
  console.error("❌ Error:", error);
  Deno.exit(1);
});
